### ISSUE TITLE
[Bug] MSAL throws a bad exception on AcquireTokenSilent + Broker if RT is expired

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1938]
**Which Version of MSAL are you using ?**
Note that to get help, you need to run the latest version. Preview version are also ok. 
For ADAL, please log issues to https://github.com/AzureAD/azure-activedirectory-library-for-dotnet
4.16

**Platform**
xamarin.android

**What authentication flow has the issue?**
AquireTokenSilent + broker

**Repro**

1. Configure the "sign in frequency" (e.g. to 1h to repro the issue easily)
2. Login interactively using broker
3. Wait 1h
4. Login silently 

**Expected behavior**
MSAL should throw MsalUiRequired exception so that apps can call AcquireTokenInteractive

**Actual behavior**
A generic "something went bad but we don't know what" MsalServiceException occured which typically crashes the apps.

**Workaround**

```csharp
try
{
    AcquireTokenSilent(...)
} 
catch (MsalServiceException e)
{
   if (e is MsalUiRequiredException || 
       e.ErrorCode == "broker_response_returned_error " ) // workaround for this bug
   {
       AcquireTokenInteractive();
   }

}

```

**Comment by bgavrilMS** [comment]
Setting as P1 as 2 customers are impacted by this.

**Comment by neha-bhargava** [comment]
Fixed in [MSAL 4.17.0 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.17.0)
